client/helper.go

```diff
@@ -219,3 +224,76 @@ func AskForConfirmation(prompt string, tries int) bool {
 	}
 	return false
 }
+
+func ReadInteger(prompt string, tries int, min int, max int) (bool, int) {
+	reader := bufio.NewReader(os.Stdin)
+	if tries <= 0 {
+		tries = 2
+	}
+
+	for ; tries > 0; tries-- {
+		fmt.Printf("%s [%d-%d]: ", prompt, min, max)
+
+		res, err := reader.ReadString('\n')
+		if err != nil {
+			log.Fatalf("error: %v", err)
+			continue
+		}
+
+		res = strings.TrimSpace(res)
+		if len(res) == 0 {
+			continue
+		}
+
+		val, err := strconv.Atoi(res)
+		if err != nil || val < min || val > max {
+			fmt.Printf("Invalid value. Must be between %d and %d.\n", min, max)
+			continue
+		}
+		return true, val
+	}
+
+	return false, -1
+}
+
+func ReadPassword(repeat bool) (string, error) {
```

Comment:
Uh quite a bit of complexity with all that terminal reset. Is that really necessary?

We don't have it in `AskForConfirmation` :thinking:

Reply:
It's not needed there as it doesn't change terminal options. But `term.ReadPassword` does, so if you press ctrl+s while entering password, you end up with terminal that doesn't echo anything on typing, until you (blindly) type `reset`.

Reply:
Ah make sense, we need to turn off echoing to hide the password. All clear now :+1: .